phenotrex/ml/shap_handler.py

```diff
@@ -0,0 +1,248 @@
+from pathlib import Path
+from typing import Tuple, Union, List
+
+import pandas as pd
+import numpy as np
+import shap
+from matplotlib import pyplot as plt
+
+from phenotrex.ml.trex_classifier import TrexClassifier
+
+
+class ShapHandler:
+    """
+    This class handles feature arrays and shap values of predictions made with phenotrex,
+    and enables plotting of shap values and summaries.
+    """
+    @classmethod
+    def from_clf(cls, clf: TrexClassifier):
+        fn, fn_idx = zip(*clf.pipeline.named_steps["vec"].get_feature_names())
+        fn = np.array(fn)
+        used_fn = [k for k, v in clf.get_feature_weights().items() if v != 0]
+        used_idxs = np.where(np.isin(fn, used_fn))[0]
+        return cls(fn, used_idxs)
+
+    def __init__(self, fn: np.ndarray, used_idxs: np.ndarray):
```

Comment:
For public scope, variables should definitely have more telling names. `feature_names`.

Reply:
The default constructor is not generally used but I concede the point, changed :)